Gulpfile.js

```diff
@@ -92,33 +94,182 @@ function buildBabel(exclude) {
+      // the version number.
+      if (process.env.CIRCLE_PR_NUMBER) {
+        const prVersion = "+pr." + process.env.CIRCLE_PR_NUMBER;
+        babelVersion += prVersion;
+        version += prVersion;
+      }
+
+      let letMinifyArgs = [];
+      if (minify) {
+        letMinifyArgs = [
+          process.env.CI ? [] : uglify(),
+          rename({ extname: ".min.js" }),
+          gulp.dest(path.join(pkg, dest)),
+        ];
+      }
+      return pump(
+        rollup({
+          rollup: require("rollup"),
+          input: getIndexFromPackage(pkg),
+          output: {
+            format,
+            name,
+          },
+          external: ["fs", "path"],
+          plugins: [
+            {
+              name: "babel-source",
+              load(id) {
+                const matches = id.match(/packages\/(babel-[^/]+)\/src\//);
+                if (matches) {
+                  // check if browser field exists for this file and replace
+                  const packageFolder = path.join(
+                    __dirname,
+                    "packages",
+                    matches[1]
+                  );
+                  const packageJson = require(path.join(
+                    packageFolder,
+                    "package.json"
+                  ));
+
+                  if (
+                    packageJson["browser"] &&
+                    typeof packageJson["browser"] === "object"
+                  ) {
+                    for (let nodeFile in packageJson["browser"]) {
+                      const browserFile = packageJson["browser"][
+                        nodeFile
+                      ].replace(/^(\.\/)?lib\//, "src/");
+                      nodeFile = nodeFile.replace(/^(\.\/)?lib\//, "src/");
+                      if (id.endsWith(nodeFile)) {
+                        if (browserFile === false) {
+                          console.log(`Ignore ${nodeFile}`);
+                          return "";
+                        }
+                        console.log(`Replace ${nodeFile} with ${browserFile}`);
+                        return fs.readFileSync(
+                          path.join(packageFolder, browserFile),
+                          "UTF-8"
+                        );
+                      }
+                    }
+                  }
+                }
+              },
+              resolveId(importee) {
+                const matches = importee.match(/^@babel\/([^/]+)$/);
+                if (matches) {
+                  // resolve babel package names to their src index file
+                  const packageFolder = path.join(
+                    __dirname,
+                    "packages",
+                    `babel-${matches[1]}`
+                  );
+                  const packageJson = require(path.join(
+                    packageFolder,
+                    "package.json"
+                  ));
+
+                  const filename =
```

Comment:
does any of our packages use `browser` field that should be resolved here? `rollup-plugin-node-resolve` handles `browser` field if u pass `browser: true` as an option to it

Reply:
The problem is that the browser field is using /lib/ and not /src/ so the node plugin won't pick it up.

Reply:
Oh, ok - I've searched for a simple "browser" fields with string value, I see we have some object maps there in i.e. `@babel/core`.

I think it might be possible to workaround this easier - rename our browser files to have `.browser.js` extension and make `rollup-plugin-node-resolve` to pick it up instead of the original one by passing `extensions: ['.browser.js', '.js']` option

Reply:
I'd assume it is fine for the `browser` field map both `src` and `lib` for Babel at least. Would that work? e.g.
```
  "browser": {
    "./lib/config/files/index.js": "./lib/config/files/index-browser.js",
    "./lib/transform-file.js": "./lib/transform-file-browser.js",
    "./lib/transform-file-sync.js": "./lib/transform-file-sync-browser.js"
    "./src/config/files/index.js": "./src/config/files/index-browser.js",
    "./src/transform-file.js": "./src/transform-file-browser.js",
    "./src/transform-file-sync.js": "./src/transform-file-sync-browser.js"
  },
```

Reply:
yeah, this definitely should work too 👍